airbyte-cdk/python/airbyte_cdk/sources/streams/checkpoint/checkpoint_reader.py

```diff
@@ -143,7 +143,7 @@ def _find_next_slice(self) -> StreamSlice:
                 # current_slice is None represents the first time we are iterating over a stream's slices. The first slice to
                 # sync not been assigned yet and must first be read from the iterator
                 next_slice = self.read_and_convert_slice()
-                state_for_slice = self._cursor.select_state(self.current_slice)
+                state_for_slice = self._cursor.select_state(next_slice)
```

Comment:
I noticed this small bug which was only found during substreams. We need to use the next slice becauase the current one is None since it is the first iteration. But decided to fix it here.

Reply:
This is actually related to the Low-Code situation in a similar manner, when we want to use the SubstreamResumableFullRefresh approach for substreams. Thanks @brianjlai for fixing it here.